client/shaders/object_shader/opengl_fragment.glsl

```diff
@@ -343,36 +474,62 @@ void main(void)
 	color = base.rgb;
 	vec4 col = vec4(color.rgb, base.a);
 	col.rgb *= varColor.rgb;
-	col.rgb *= emissiveColor.rgb * vIDiff;
+	col.rgb *= vIDiff;
 
 #ifdef ENABLE_DYNAMIC_SHADOWS
 	float shadow_int = 0.0;
 	vec3 shadow_color = vec3(0.0, 0.0, 0.0);
 	vec3 posLightSpace = getLightSpacePosition();
 
+	float distance_rate = (1 - pow(clamp(2.0 * length(posLightSpace.xy - 0.5),0.0,1.0), 20.0));
+	float f_adj_shadow_strength = max(adj_shadow_strength-mtsmoothstep(0.9,1.1,  posLightSpace.z  ),0.0);
+
+	if (distance_rate > 1e-7) {
+	
 #ifdef COLORED_SHADOWS
-	vec4 visibility;
-	if (cosLight > 0.0)
-		visibility = getShadowColor(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
-	else
-		visibility = vec4(1.0, 0.0, 0.0, 0.0);
-	shadow_int = visibility.r;
-	shadow_color = visibility.gba;
+		vec4 visibility;
+		if (cosLight > 0.0)
+			visibility = getShadowColor(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
+		else
+			visibility = vec4(1.0, 0.0, 0.0, 0.0);
+		shadow_int = visibility.r;
+		shadow_color = visibility.gba;
 #else
-	shadow_int = getShadow(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
+		if (cosLight > 0.0)
+			shadow_int = getShadow(ShadowMapSampler, posLightSpace.xy, posLightSpace.z);
+		else
+			shadow_int = 1.0;
 #endif
+		shadow_int *= distance_rate;
+		shadow_int = clamp(shadow_int, 0.0, 1.0);
 
-	if (f_normal_length != 0 && cosLight <= 0.001) {
-		shadow_int = clamp(shadow_int + 0.5 * abs(cosLight), 0.0, 1.0);
 	}
 
-	shadow_int = 1.0 - (shadow_int * adj_shadow_strength);
+	// turns out that nightRatio falls off much faster than
+	// actual brightness of artificial light in relation to natual light.
+	// Power ratio was measured on torches in MTG (brightness = 14).
+	float adjusted_night_ratio = pow(nightRatio, 0.6);
+
+	// cosine of the normal-to-light angle when
+	// we start to apply self-shadowing
+	const float self_shadow_cutoff_cosine = 0.14;
+	if (f_normal_length != 0 && cosLight < self_shadow_cutoff_cosine) {
+		shadow_int = max(shadow_int, 1 - clamp(cosLight, 0.0, self_shadow_cutoff_cosine)/self_shadow_cutoff_cosine);
+		shadow_color = mix(vec3(0.0), shadow_color, min(cosLight, self_shadow_cutoff_cosine)/self_shadow_cutoff_cosine);
+	}
 
-	col.rgb = mix(shadow_color, col.rgb, shadow_int) * shadow_int;
+	shadow_int *= f_adj_shadow_strength;
+	
+	// calculate fragment color from components:
+	col.rgb =
+			adjusted_night_ratio * col.rgb + // artificial light
+			(1.0 - adjusted_night_ratio) * ( // natural light
+					col.rgb * (1.0 - shadow_int * (1.0 - shadow_color)) +  // filtered texture color
+					dayLight * shadow_color * shadow_int);                 // reflected filtered sunlight/moonlight
+	// col.r = 0.5 * clamp(getPenumbraRadius(ShadowMapSampler, posLightSpace.xy, posLightSpace.z, 1.0) / SOFTSHADOWRADIUS, 0.0, 1.0) + 0.5 * col.r;
```

Comment:
Let's just remove the commented stuff.

Reply:
Removed from both node and object shaders.